ecl/hql/hqlmeta.cpp

```diff
@@ -2443,6 +2443,22 @@ void calculateDatasetMeta(CHqlMetaInfo & meta, IHqlExpression * expr)
             }
             break;
         }
+    case no_quantile:
+        //It is not be safe to assume that all implementations of the quantile activity will generate their results in order.
+        if (expr->hasAttribute(localAtom) || isGrouped(dataset))
+        {
+            IHqlExpression * transform = expr->queryChild(3);
+            OwnedHqlExpr leftSelect = createSelector(no_left, dataset, expr->queryAttribute(_selectorSequence_Atom));
+            TableProjectMapper mapper;
+            mapper.setMapping(transform, leftSelect);
+
+            extractMeta(meta, dataset);
+            meta.removeAllSortOrders();
+            meta.applyProject(mapper);  // distribution preserved if local..
+        }
+        else
+            meta.preserveGrouping(dataset);
```

Comment:
Something seemed odd (semanticly) given the previous isGrouped(dataset), looking, I see that preserveGrouping also checks this.

In addition CHqlMetaInfo::clearGrouping() also clears groupSortOrder, should preserveGrouping do so also and thus just call clearGrouping()?